[data] Support CountDistinct aggregate

my-vegetable-has-exploded opened the pull request:
## Description

`CountDistinct` allow users to compute the number of distinct values in a column, similar to SQL's `COUNT(DISTINCT ...)`. 

## Related issues

close #58252

## Additional information
> Optional: Add implementation details, API changes, usage examples, screenshots, etc.

my-vegetable-has-exploded committed:
feat(data): support CountDistinct aggregate.

Signed-off-by: my-vegetable-has-exploded <wy1109468038@gmail.com>

my-vegetable-has-exploded committed:
fix typos.

Signed-off-by: my-vegetable-has-exploded <wy1109468038@gmail.com>

my-vegetable-has-exploded committed:
fix nunique arg.

Signed-off-by: my-vegetable-has-exploded <wy1109468038@gmail.com>

alexeykudinkin reviewed (commented):
LGTM, please address feedback from Gemini

my-vegetable-has-exploded committed:
Merge remote-tracking branch 'origin/master' into count-distinct

Signed-off-by: my-vegetable-has-exploded <wy1109468038@gmail.com>

my-vegetable-has-exploded committed:
fix ignore_nulls.

Signed-off-by: my-vegetable-has-exploded <wy1109468038@gmail.com>

goutamvenkat-anyscale committed:
Merge branch 'master' into count-distinct

my-vegetable-has-exploded commented:
> > Hi @alexeykudinkin, thanks for review. I have resolved problems, but ci fails. I don't find actual fail info in microcheck. Could you please re-triage this test？
> 
> Some of the errors were transient (5xx errors). I just merge master into your branch. Those should resolve

Thanks @goutamvenkat-anyscale !!!

goutamvenkat-anyscale commented:
@my-vegetable-has-exploded 

[2026-01-01T21:27:19Z] Public APIs that are NOT documented:
--
[2026-01-01T21:27:19Z] 	ray.data.aggregate.CountDistinct


Looks like you're missing some documentation on `CountDistinct` which is the root cause of the CI failure

my-vegetable-has-exploded committed:
docs(api): add CountDistinct to aggregate.rst API reference

Adds CountDistinct to the list of aggregation functions in the API reference
to improve documentation completeness.

Signed-off-by: my-vegetable-has-exploded <wy1109468038@gmail.com>

my-vegetable-has-exploded commented:
> @my-vegetable-has-exploded
> ## [2026-01-01T21:27:19Z] Public APIs that are NOT documented:
> 
> [2026-01-01T21:27:19Z] ray.data.aggregate.CountDistinct
> 
> Looks like you're missing some documentation on `CountDistinct` which is the root cause of the CI failure

Done
